python/common/org/python/types/Str.java

```diff
@@ -764,7 +764,20 @@ private long toPositiveIndex(long index) {
             __doc__ = ""
     )
     public org.python.Object isnumeric() {
-        throw new org.python.exceptions.NotImplementedError("isnumeric() has not been implemented.");
+        int c = 0;
+        java.lang.String checkString = this.value;
+        char currentCharacter;
+        if (checkString.length() == 0) {
+            return new org.python.types.Bool(false);
+        }
+        while (c < (checkString.length())) {
```

Comment:
It might be clearer to have this as a `for (int c = 0; c < checkString.length(); c++)` loop instead, since you aren't using the value of `c` after and outside the loop.